### ISSUE TITLE
0.740 doesn't understand abstract properties

### ISSUE THREAD

**Issue by hukkinj1** [python/mypy#7760]
Running mypy 0.740 with `check_untyped_defs = True` gives a false positive
`mypy_bug.py:12: error: "Callable[[AbstractBase], Dict[Any, Any]]" has no attribute "values"`
when running against the code below:

```python
from abc import ABC, abstractmethod


class AbstractBase(ABC):
    @property
    @abstractmethod
    def SOME_DICT(self) -> dict:
        pass

    @classmethod
    def get_dict_values(cls):
        return cls.SOME_DICT.values()


class Implementation(AbstractBase):
    SOME_DICT = {"some": "data"}
```

Mypy thinks that the value of `cls.SOME_DICT` on line 12 is `Callable[[AbstractBase], Dict[Any, Any]]`, even though it is a property, typehinted to be `dict`.

**Comment by ilevkivskyi** [comment]
Mypy is correct and your code is unsafe (although the error message is a bit cryptic), properties should not be accessed on class objects. Consider:
```python
class Implementation(AbstractBase):
     @property
     def SOME_DICT(self) -> dict:
         return {"some": "data"}

Implementation.det_dict_values()
```
This code fails with a traceback (similar to what mypy says):
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 8, in get_dict_values
AttributeError: 'property' object has no attribute 'values'
```

**Comment by hukkinj1** [comment]
Thanks, understood. I guess my real issue then is Python language's lack of notation for abstract class attributes...
